inference/worker/utils.py

```diff
@@ -38,3 +41,20 @@ def finish(self, reason: Literal["length", "eos_token", "stop_sequence"]):
             yield from self.tokens
         else:
             yield from self.tokens
+
+
+def wait_for_inference_server(inference_server_url: str, timeout: int = 600):
+    health_url = f"{inference_server_url}/health"
+    time_limit = time.time() + timeout
+    while True:
+        try:
+            response = requests.get(health_url)
+            response.raise_for_status()
+        except (requests.HTTPError, requests.ConnectionError):
+            if time.time() > time_limit:
+                raise
+            logger.warning("Inference server not ready. Retrying...")
+            time.sleep(5)
```

Comment:
Maybe we could randomize this delay to de-synchronize client retry requests. Otherwise all clients will try to reconnect at a similar time in case of a server restart etc.

Reply:
good idea